### PR TITLE
Compact the HelpView.

### PR DESCRIPTION
**Pull request by matrixise**
Define the height attribute (default to the number of key bindings) and
remove the bottom line character.

### CODE REVIEW

**Comment by matrixise** [comment]
https://github.com/zulip/zulip-terminal/issues/297

**Comment by amanagr** [comment]
@neiljp this look like a good idea to me. What do you think?

**Comment by matrixise** [comment]
with a 25" screen, when your terminal is full, you see a table with a lot of dashed lines :/ I prefer a small box with the essential.

File: zulipterminal/core.py [hunk 1]
```diff
@@ -85,14 +85,15 @@ def draw_screen(self, *args: Any, **kwargs: Any) -> None:
 
     def show_help(self) -> None:
         help_view = HelpView(self)
+        dialog_height = min(50, help_view.height)
```
Comment (rht) [inline zulipterminal/core.py]: 50 is probably too compact. It displays all the way to 'Scroll to bottom', whereas the current version (with '-') displays to 'Send a message'. It takes <1/3 of my vertical screen space on a 12.5" screen.
2019-02-12-123859_1522x767_scrot

Comment (matrixise) [reply zulipterminal/core.py]: @rht I just updated, using the height of the screen and taking the half.

Comment (rht) [reply zulipterminal/core.py]: Yeah looks better / less cramped to me.

Comment (rht) [reply zulipterminal/core.py]: Incidentally it is also 50% of screen on webapp, though not sure if it continues to stretch to be 50% on larger screen,

File: zulipterminal/ui_tools/views.py [hunk 2]
```diff
@@ -571,6 +571,7 @@ def __init__(self, controller: Any) -> None:
                   for binding in KEY_BINDINGS.values()]
         max_widths = [max(width) for width in zip(*widths)]
         self.width = sum(max_widths)
+        self.height = len(KEY_BINDINGS)
```
[the remark this answers is not in the document]
Comment (matrixise) [reply zulipterminal/ui_tools/views.py]: sure, sorry, I have just pushed a new revision.

**Comment by matrixise** [comment]
@rht Tell me if you prefer one commit (squash&merge) or many commits for this PR.

**Comment by rht** [comment]
Usually, it is many commits within the theme of the PR, but each commits are 'correct' in that it doesn't have to be fixed by the next commit(s). i.e. in this case you should squash them.

**Comment by matrixise** [comment]
fixed, just rebased the code and merge the 3 commits in only one.

File: zulipterminal/ui_tools/views.py [hunk 3]
```diff
@@ -580,7 +580,7 @@ def __init__(self, controller: Any) -> None:
                     (max_widths[1], urwid.Text(", ".join(binding['keys'])))
                               ], dividechars=2),
                 tlcorner='', brcorner='', trcorner='', blcorner='',
-                rline=' ', lline=' ', bline='-', tline='',
```
Comment (neiljp) [inline zulipterminal/ui_tools/views.py]: If we're removing this, I'm not sure what the point of the `LineBox` is now?

**Comment by rht** [comment]
(Just want to additionally comment that without the dashes it looks smoother / less command-line-y (you see dashes in ASCII art).)

**Comment by rht** [comment]
IMO, compact with alternating background color > compact (this PR) >  with dashes (current master). I think it is fine as a transition state?

File: zulipterminal/ui_tools/views.py (hunk 3 above)
Comment (rht) [reply zulipterminal/ui_tools/views.py]: As in, L577-585 are to be removed?

Comment (neiljp) [reply zulipterminal/ui_tools/views.py]: I meant that the `LineBox` 'wrapper' around the `Columns` is rather less useful now; there is a left- and right- padding there, but I think that would be easier as a padding around the entire 'table' (rather than each entry), or just as spaces to the left and right of each entry.

**Comment by neiljp** [comment]
TLDR: Given @amanagr 's comment, I'm going to merge now, but with the following concerns/caveats:
* we've definitely lost some action-to-key clarity with removing the lines
* half height + no lines shows maybe 3 fewer entries? So it's more compact, but less obvious?
(though, maybe a small 'window' helps make the absent lines clearer?)
* the linebox issue is not major & I think we can just remove the side spaces now too
* the referring issue has screenshots of 0.3.1, not master, so it's not a direct comparison there.

I have a provisional solution to most of these issues, so would encourage those who were interested in this change to review that upcoming PR.

This change is all the more relevant assuming we plan to integrate the readline keys into the online help, as that will add even more actions/keys to the list!
